x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -911,14 +923,45 @@ export class ObservabilityAIAssistantClient {
     return this.dependencies.knowledgeBaseService.deleteEntry({ id });
   };
 
-  private addResponseLanguage = (messages: Message[], responseLanguage: string): Message[] => {
+  private resolveInstructions = async (requestInstructions: UserInstruction[]) => {
+    const knowledgeBaseInstructions =
+      await this.dependencies.knowledgeBaseService.getInstructions();
+
+    if (requestInstructions.length + knowledgeBaseInstructions.length === 0) {
+      return '';
+    }
+
+    const knowledgeBaseInstructionsById = knowledgeBaseInstructions.reduce(
+      (acc, next) => ({
+        ...acc,
+        [next.doc_id]: next.text,
+      }),
+      {}
+    );
+
+    const instructionsById = requestInstructions.reduce(
```

Comment:
same here - can we avoid the new object? why do we need the ids here anyway?

Reply:
> why do we need the ids here anyway?

See https://github.com/elastic/kibana/pull/180263#discussion_r1555688722